src/components/SearchForm.jsx

```diff
@@ -1,60 +1,78 @@
 import React from 'react';
-import { Formik, Field, Form, withFormik } from 'formik';
+import { Field, Form, withFormik } from 'formik';
+import * as Yup from 'yup';
 
 const SearchForm = ({
   values,
-  handleChange,
-  handleSubmit
+  errors,
+  touched,
 }) => (
-  <form
-    onSubmit={handleSubmit}
-  >
-    <label>Include these words or phrases</label>
-    <input
-      type="text"
-      name="searchTerms"
-      placeholder="seperate search terms with a comma (,)"
-      value={values.searchTerms}
-      onChange={handleChange}
-    />
+  <Form>
+    <div className="form-field">
+      <label>Include these words or phrases
+        <Field
+          type="text"
+          name="searchTerms"
+          placeholder="seperate search terms with a comma (,)"
+        />
+      </label>
+      {touched.searchTerms && errors.searchTerms &&
+      <p className="form-error">{errors.searchTerms}</p>}
+    </div>
+    <div className="form-field">
+      <label>Location (optional, zipcode)
+        <Field
+          type="text"
+          name="location"
+          placeholder="97015"
+        />
+      </label>
+      {touched.location && errors.location &&
+      <p className="form-error">{errors.location}</p>}
+    </div>
+    <div className="form-field">
+      <label>radius (in miles)
+        <Field
+          type="text"
+          name="radius"
+          placeholder="25"
+        />
+      </label>
+      {touched.location && errors.radius &&
+      <p className="form-error">{errors.radius}</p>}
+    </div>
+    <div className="form-field">
+      <label>Timeframe (optional)
+        <Field
+          type="text"
+          name="timeFrame"
+          placeholder="1 month"
+        />
+      </label>
+      {touched.timeFrame && errors.timeFrame &&
+      <p className="form-error">{errors.timeFrame}</p>}
+    </div>
+    <label>
+      <Field
+        type="checkbox" name="savedSearch" checked={values.savedSearch} />
+      Save this search?
+    </label>
     <br/>
-    <label>Location (optional, zipcode)</label>
-    <input
-      type="text"
-      name="location"
-      placeholder="97015"
-      value={values.location}
-      onChange={handleChange}
-    />
-    <br/>
-    <label>radius (in miles)</label>
-    <input
-      type="text"
-      name="radius"
-      placeholder="25"
-      value={values.radius}
-      onChange={handleChange}
-    />
-    <br/>
-    <label>Timeframe (optional)</label>
-    <input
-      type="text"
-      name="timeFrame"
-      placeholder="1 month"
-      value={values.timeFrame}
-      onChange={handleChange}
-    />
+    <Field component="select" name="plan">
+      <option value="free">Free</option>
+      <option value="premium">Premium</option>
+    </Field>
```

Comment:
Why is this element part of the form?  I don't remember it being part of any of our mock-ups previously.

Reply:
I agree that this seems odd.  Was it part of an example that you forgot to remove?